akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Compliance.scala

```diff
@@ -25,10 +25,20 @@ private[akka] object Http2Compliance {
 
   // @DoNotInherit
   private[akka] sealed class IllegalHttp2FrameSize(size: Int, expected: String)
-    extends IllegalArgumentException(s"Illegal HTTP/2 frame size: [$size]. $expected!")
+    extends Http2ProtocolException(ErrorCode.FRAME_SIZE_ERROR, s"Illegal HTTP/2 frame size: [$size]. $expected!")
 
   // require methods use `if` because `require` allocates
 
+  /** Validate value of MAX_FRAME_SIZE setting. */
+  def validateMaxFrameSize(value: Int): Unit = {
+    import Http2Protocol.MinFrameSize
+    import Http2Protocol.MaxFrameSize
+    if (value < MinFrameSize) throw new Http2ProtocolException(ErrorCode.FRAME_SIZE_ERROR, s"MAX_FRAME_SIZE MUST NOT be < than $MinFrameSize, attempted setting to: $value!")
```

Comment:
6.5.2:

      Values outside this range MUST be treated as a connection error
      (Section 5.4.1) of type PROTOCOL_ERROR.